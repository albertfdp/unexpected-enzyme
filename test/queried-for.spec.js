import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React from 'react';

class Fixture extends React.Component {
  render() {
    return (
      <div>
        <div id="parent">
          <div id="child">Children</div>
        </div>
      </div>
    );
  }
}

describe('queried-for', () => {
  it('passes when the actual matches the expected', () => {
    const wrapper = mount(<Fixture />);

    expect(
      wrapper,
      'queried for',
      '#parent',
      'to render as',
      <div id="child">Children</div>
    );
  });

  it('fails when the actual does not match the expected', () => {
    const wrapper = mount(<Fixture />);

    expect(
      () =>
        expect(
          wrapper,
          'queried for',
          '#parent',
          'to render as',
          <div id="child">Foo</div>
        ),
      'with error matching snapshot'
    );
  });
});
