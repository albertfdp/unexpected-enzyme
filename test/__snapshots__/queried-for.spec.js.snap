// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queried-for fails when the actual does not match the expected 1`] = `
"expected <Fixture /> queried for '#parent' to render as <div id=\\"child\\">Foo</div>

<div id=\\"parent\\">
  <div id=\\"child\\">
    Children // -Children
             // +Foo
  </div>
</div>"
`;
